hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -643,20 +660,21 @@ HFileBlock unpack(HFileContext fileContext, FSReader reader) throws IOException
    * Always allocates a new buffer of the correct size. Copies header bytes from the existing
    * buffer. Does not change header fields. Reserve room to keep checksum bytes too.
    */
-  private void allocateBuffer() {
+  private ByteBuff allocateBufferForUnpacking() {
```

Comment:
This method used to be called after creating a new block, and creating the new block did a `buf.duplicate()` in createBuilder. Now we call this method _before_ creating the new block, so we call buf.duplicate() below instead. Same operations as before, just a different order so that we can examine the resulting newBuf in creating the new block.